File: Tutorial_Minggu_3/Rekening.cpp
#include "Rekening.h"

Rekening::Rekening(double d) {
	if (d < 0) d = 0;
	saldo = d;
}

void Rekening::setSaldo(double d) {
	saldo = d;
}

double Rekening::getSaldo() const {
	return saldo;
}

void Rekening::simpanUang(double d) {
	saldo += d;
}

bool Rekening::tarikUang(double d) {
	if (saldo < d) return false;
	saldo -= d;
	return true;	
}

File: Tutorial_Minggu_7/rekening/Rekening.java
abstract class Rekening {
	protected String nama;
	protected double saldo;
	protected double sukuBunga;

	public Rekening(String n, double d, double s) {
		this.nama = n;
		this.saldo = (d < 0 ? 0 : d);
		this.sukuBunga = s;
	}

	abstract public void setor(double uang);
	abstract public void tarik(double uang);
	abstract public void update();

	public double hitungBiaya() {
		double x1 = 10.0;
		double x2 = 0.1 * (this.saldo);
		return (x1 < x2 ? x1 : x2);
	}

	public String getNama() {
		return this.nama;
	}

	public double getSaldo() {
		return this.saldo;
	}

	public double getSukuBunga() {
		return this.sukuBunga;
	}
}